src/Microsoft.OData.Core/ODataBatchReader.cs

```diff
@@ -270,274 +263,25 @@ private Task<bool> ReadAsynchronously()
-        private ODataBatchOperationRequestMessage CreateOperationRequestMessageImplementation()
-        {
-            this.operationState = OperationState.MessageCreated;
-
-            string requestLine = this.batchStream.ReadFirstNonEmptyLine();
-
-            string httpMethod;
-            Uri requestUri;
-            this.ParseRequestLine(requestLine, out httpMethod, out requestUri);
-
-            // Read all headers and create the request message
-            ODataBatchOperationHeaders headers = this.batchStream.ReadHeaders();
-
-            if (this.batchStream.ChangeSetBoundary != null)
-            {
-                if (this.allowLegacyContentIdBehaviour)
-                {
-                    // Add a potential Content-ID header to the URL resolver so that it will be available
-                    // to subsequent operations.
-                    string contentId;
-                    if (this.contentIdToAddOnNextRead == null && headers.TryGetValue(ODataConstants.ContentIdHeader, out contentId))
-                    {
-                        if (contentId != null && this.urlResolver.ContainsContentId(contentId))
-                        {
-                            throw new ODataException(Strings.ODataBatchReader_DuplicateContentIDsNotAllowed(contentId));
-                        }
-
-                        this.contentIdToAddOnNextRead = contentId;
-                    }
-                }
-
-                if (this.contentIdToAddOnNextRead == null)
-                {
-                    throw new ODataException(Strings.ODataBatchOperationHeaderDictionary_KeyNotFound(ODataConstants.ContentIdHeader));
-                }
-            }
-
-            ODataBatchOperationRequestMessage requestMessage = ODataBatchOperationRequestMessage.CreateReadMessage(
-                this.batchStream,
-                httpMethod,
-                requestUri,
-                headers,
-                /*operationListener*/ this,
-                this.contentIdToAddOnNextRead,
-                this.urlResolver);
-
-            return requestMessage;
-        }
-
         /// <summary>
-        /// Returns the cached <see cref="ODataBatchOperationRequestMessage"/> for reading the content of an operation 
-        /// in a batch request.
+        /// Parses the request line of a batch operation request, without HTTP method validation.
         /// </summary>
-        /// <returns>The message that can be used to read the content of the batch request operation from.</returns>
-        private ODataBatchOperationResponseMessage CreateOperationResponseMessageImplementation()
+        /// <param name="requestLine">The request line as a string.</param>
+        /// <param name="httpMethod">The parsed HTTP method of the request.</param>
+        /// <param name="requestUri">The parsed <see cref="Uri"/> of the request.</param>
+        internal void ParseRequestLine(string requestLine, out string httpMethod, out Uri requestUri)

```

Comment:
wrapper method removed; it was some leftover from removed impl. Also moved the wrapped method to ODataMultipartMixedBatchReader, which becomes the only caller.